include/gabia/secure_socket.hpp

```diff
@@ -0,0 +1,394 @@
+//
+// Copyright (c) 2017 Damian Jarek (damian dot jarek93 at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/djarek/gabia
+//
+
+#ifndef GABIA_SECURE_SOCKET_HPP
+#define GABIA_SECURE_SOCKET_HPP
+
+#include <gabia/bytes.hpp>
+#include <gabia/crypto/aead.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/beast.hpp>
+#include <boost/endian/arithmetic.hpp>
+
+#define CORO_YIELD BOOST_ASIO_CORO_YIELD
+#define CORO_REENTER BOOST_ASIO_CORO_REENTER
+
+namespace gabia {
+
+namespace asio = boost::asio;
+namespace beast = boost::beast;
```

Comment:
I know it sucks to have to write out "boost::beast::" all the time but you are effectively making `gabia::asio` and `gabia::beast` part of your **public** interface. Is this really what you want?